src/message.rs

```diff
@@ -90,6 +90,16 @@ impl StructuredData {
         }
     }
 
+    /// Fetch or insert a new sd_id entry into the StructuredData
+    pub fn entry<SI>(&mut self, sd_id: SI) -> &mut BTreeMap<String, String>
```

Comment:
Let's DRY stuff up and change the `.insert_tuple()` method to call `.entry()`

Also, it looks like rustfmt doesn't like your indentation. I don't have it set up as a lint yet, but if you wouldn't mind running `cargo fmt`, that'll avoid reformatting next time I edit the file.

Reply:
done